src/basic/socket-util.c

```diff
@@ -648,12 +648,7 @@ int sockaddr_pretty(const struct sockaddr *_sa, socklen_t salen, bool translate_
                                 return -ENOMEM;
 
                 } else if (sa->un.sun_path[0] == 0) {
-                        /* abstract */
-
-                        /* FIXME: We assume we can print the
-                         * socket path here and that it hasn't
-                         * more than one NUL byte. That is
-                         * actually an invalid assumption */
+                        /* Abstract socket */
```

Comment:
well, we make no restrictions on how the "struct sockaddr" came to be. While we are doing this properly it doesn't mean everybody does. 

I would feel more comfortable if we'd simply generate an error if there's an embedded NUL in this case then simply assume it doesn't happen. i.e. return EADDRNOTAVAIL in if there's any NUL char in sun_path that is not at the end?